apps/api/src/app/layouts/use-cases/check-layout-is-used/check-layout-is-used.use-case.ts

```diff
@@ -0,0 +1,25 @@
+import { LayoutEntity, LayoutRepository } from '@novu/dal';
+import { Injectable } from '@nestjs/common';
+
+import { CheckLayoutIsUsedCommand } from './check-layout-is-used.command';
+import {
+  FindMessageTemplatesByLayoutCommand,
+  FindMessageTemplatesByLayoutUseCase,
+} from '../../../message-template/usecases';
+
+@Injectable()
+export class CheckLayoutIsUsedUseCase {
```

Comment:
I guess we could quite easily use this for the isUsed in the FE table we were planning to scope out  🤭
Maybe we could just do it as a query in the front or as an extra return field to the get-layout-usecase , instead of actually adding a field to the schema. Which I definitely don't like.  

Actually might have some value in also making it external API accessible.

Reply:
I do agree that the `isUsed` calculation should be done on demand and not by DB flags that could go out of sync easily. And with the rest of your comments.